### PR TITLE
Leverage improvements in SQLKit and SQLiteNIO

### PR DESCRIPTION
**Pull request by gwynne**
Several improvements:

- Adds full `Sendable` correctness, including taking advantage of the improvements in vapor/sqlite-nio#68
- Adds `ExistentialAny` compliance
- The minimum Swift version is now 5.8
- Fully supports the new SQLKit functionality added in [SQLKit 3.29.0](https://github.com/vapor/sql-kit/releases/tag/3.29.0).
- Modernizes the README and API documentation. API docs now have 100% coverage.
- Custom JSON encoders and decoders can now be specified per-connection.
- SQLKit's `queryLogLevel` functionality is now fully implemented.
- Database connections now default to using `NIOThreadPool.singleton` and `MultiThreadedEventLoopGroup.singleton` unless otherwise specified.
- Leverages the improvements from vapor/sqlite-nio#68 to improve performance of Concurrency-based APIs by reducing thread hops and excess allocations.
- `SQLiteDataEncoder` and `SQLiteDataDecoder` are now slightly faster and behave more consistently.

### CODE REVIEW

File: Package.swift
Comment (0xTim) [inline Package.swift:47]: This won't work in the 5.8 manifest

File: Sources/SQLiteKit/SQLiteConnection+SQLKit.swift
Comment (0xTim) [inline Sources/SQLiteKit/SQLiteConnection+SQLKit.swift:120]: I'm going to push back on this - `SQLiteDatabase` can easily be sendable, as can the types that conform to it so I think we should do that

**Review by 0xTim — CHANGES_REQUESTED** [review]
Couple of points to discuss

File: Package.swift
Comment (gwynne) [reply Package.swift:47]: ... Now he tells me 😂

File: Sources/SQLiteKit/SQLiteConnection+SQLKit.swift
Comment (gwynne) [reply Sources/SQLiteKit/SQLiteConnection+SQLKit.swift:120]: Working on that over in the SQLiteNIO PR, but I didn't want to block this one on that.

Comment (0xTim) [reply Sources/SQLiteKit/SQLiteConnection+SQLKit.swift:120]: IMO it makes sense to get the other PR landed first with the sendable annotations and then land this. Don't feel super strongly about it though

File: Package.swift
Comment (gwynne) [reply Package.swift]: In the past I've been leaving these in the older manifest because they're no-ops, but in retrospect it really makes more sense not to pretend that features are there when they aren't.

File: Sources/SQLiteKit/SQLiteConnection+SQLKit.swift
Comment (gwynne) [reply Sources/SQLiteKit/SQLiteConnection+SQLKit.swift:120]: Yeah, that's probably better at this point. I didn't even have the SQLiteNIO PR in mind yet when I first wrote this one.

**Comment by gwynne** [comment]
Tests for this PR will fail until vapor/sqlite-nio#68 lands.

**Review by MahdiBM — APPROVED** [review]
Looks good at a high level.

**Review by 0xTim — APPROVED** [review]

**Pull request merged**
